lib/sles4sap/ipaddr2.pm

```diff
@@ -0,0 +1,223 @@
+use publiccloud::utils;
+
+
+=head1 SYNOPSIS
+
+Library to manage ipaddr2 tests
+=cut
+
+our @EXPORT = qw(
+  $user
+  ipaddr2_azure_deployment
+  ipaddr2_ssh_cmd
+  ipaddr2_destroy
+);
+
+use constant DEPLOY_PREFIX => 'ip2t';
+
+our $user = 'cloudadmin';
+our $bastion_pub_ip = DEPLOY_PREFIX . '-pub_ip';
+
+sub ipaddr2_azure_resource_group {
+    return DEPLOY_PREFIX . get_current_job_id();
+}
+
+sub ipaddr2_azure_deployment {
+    my (%args) = @_;
+    foreach (qw(region os)) {
+        croak("Argument < $_ > missing") unless $args{$_}; }
+
+    az_version();
+
+    my $rg = ipaddr2_azure_resource_group();
+
+    az_group_create(
+        name => $rg,
+        region => $args{region});
+
+    # Create a VNET only needed later when creating the VM
+    my $vnet = DEPLOY_PREFIX . '-vnet';
+    my $subnet = DEPLOY_PREFIX . '-snet';
+    my $priv_ip_range = '192.168.0';
+    az_network_vnet_create(
+        resource_group => $rg,
+        region => $args{region},
+        vnet => $vnet,
+        address_prefixes => $priv_ip_range . '.0/16',
+        snet => $subnet,
+        subnet_prefixes => $priv_ip_range . '.0/24');
+
+    # Create a Network Security Group
+    # only needed later when creating the VM
+    my $nsg = DEPLOY_PREFIX . '-nsg';
+    az_network_nsg_create(
+        resource_group => $rg,
+        name => $nsg);
+
+    # Create the only one public IP in this deployment,
+    # it will be assigned to the 3rd VM (bastion role)
+    az_network_publicip_create(
+        resource_group => $rg,
+        name => $bastion_pub_ip,
+        sku => 'Basic',
+        allocation_method => 'Static');
+
+    # Create the load balancer entity.
+    # Mostly this one is just a "group" definition
+    # to link back-end (2 VMs) and front-end (the Pub IP) resources
+    # SKU Standard (and not Basic) is needed to get some Metrics
+    my $lb = DEPLOY_PREFIX . '-lb';
+    my $lb_be = DEPLOY_PREFIX . '-backend_pool';
+    my $lb_fe = DEPLOY_PREFIX . '-frontent_ip';
+    my $lb_feip = $priv_ip_range . '.50';
+    az_network_lb_create(
+        resource_group => $rg,
+        name => $lb,
+        vnet => $vnet,
+        snet => $subnet,
+        backend => $lb_be,
+        frontend_ip => $lb_fe,
+        fip => $lb_feip,
+        sku => 'Standard');
+
+    # All the 2 VM will be later assigned to it.
+    # The load balancer does not explicitly knows about it
+    my $as = DEPLOY_PREFIX . '-as';
+    az_vm_as_create(
+        resource_group => $rg,
+        name => $as,
+        region => $args{region},
+        fault_count => 2);
+
+    # Create 2:
+    #   - VMs
+    #   - for each of them open port 80
+    #   - link their NIC/ipconfigs to the load balancer to be managed
+    my $vm;
+    my $cloud_init_file = '/tmp/cloud-init-web.txt';
+    assert_script_run(join(' ',
+            'curl -v -fL',
+            data_url('sles4sap/cloud-init-web.txt'),
+            '-o', $cloud_init_file));
+    for (my $i = 0; $i < get_var('IPADDR2_VMS', 2); $i++) {
```

Comment:
What is in this setting? The number of Virtual Machines (seems to be the case after looking at the code) or the number of IpAddr2 resources in the cluster? Perhaps the name can cause confusion.

Reply:
it is about the number of VM, that also has to be used on other kind of resources as some of the resources are per VM.

On the other hand for the moment I think I can hardcode it to `2` without any setting